### PR TITLE
Add workaround low memory crash during image saving

### PR DESCRIPTION
**Pull request by ArDiouscuros**
Sample model moved to cpu before processing samples to image
I am not sure how to handle additional parameter and supply it to underplaying classes
Related to FR #775

### CODE REVIEW

**Comment by ArDiouscuros** [comment]
With with change I could not only generate 512x512 without crash on save, but even go to 512x1024

File: ldm/generate.py [hunk 1]
```diff
@@ -649,6 +649,7 @@ def _make_txt2img(self):
         if not self.generators.get('txt2img'):
             from ldm.dream.generator.txt2img import Txt2Img
             self.generators['txt2img'] = Txt2Img(self.model, self.precision)
+            self.generators['txt2img'].free_gpu_mem = self.free_gpu_mem
```
Comment (lstein) [inline ldm/generate.py:652]: This is a bit awkward, but there's no other way to do it in the current model without adding more arguments to its __init__ method and then making the corresponding changes in generate.py.

File: scripts/dream.py [hunk 2]
```diff
@@ -108,6 +108,8 @@ def main():
 
     # preload the model
     gen.load_model()
+    #set additional option
+    gen.free_gpu_mem = opt.free_gpu_mem
```
Comment (lstein) [inline scripts/dream.py:112]: This is a bit awkward, but there's no other way to do it in the current model without adding more arguments to its __init__ method and then making the corresponding changes in dream.py.

**Review by lstein — APPROVED** [review]
Thank you. Could you also add a line to TROUBLESHOOTING.md and CLI.md that tells users about this feature? Otherwise they may not find it on their own. Also add your name to CONTRIBUTORS, if it is not already there.

**Pull request merged**